kobo/apps/audit_log/tests/api/v2/test_api_audit_log.py

```diff
@@ -106,11 +156,132 @@ def test_filter_list(self):
+        response = self.client.get(self.url)
+        # only return user1's access logs
+        self.assert_audit_log_results_equal(
+            response=response,
+            expected_kwargs={'action': AuditAction.AUTH, 'user': user1},
+        )
+
+    def test_endpoint_ignores_querystring(self):
+        # make sure a user can't get someone else's access logs
+        user1 = User.objects.get(username='someuser')
+        self.force_login_user(user1)
+        response = self.client.get(f'{self.url}?q=user__username:anotheruser')
+        # check we still only got logs for the logged-in user
+        for audit_log_dict in response.data['results']:
+            self.assertEquals(audit_log_dict['username'], 'someuser')
+
+
+class AllApiAccessLogsTestCase(BaseAuditLogTestCase):
+
+    def get_endpoint_basename(self):
+        return 'all-access-logs-list'
+
+    def setUp(self):
+        super().setUp()
+        super_user = User.objects.get(username='admin')
+        user2 = User.objects.get(username='anotheruser')
+        # generate 3 access logs, 2 for superuser, 1 for user2
+        self.create_access_log(super_user)
+        self.create_access_log(super_user)
+        self.create_access_log(user2)
+
+        # create a random non-auth audit log
+        log = AuditLog.objects.create(
+            user=User.objects.get(username='someuser'),
+            app_label='foo',
+            model_name='bar',
+            object_id=1,
+            action=AuditAction.DELETE,
+        )
+        log.save()
+        self.assertEqual(AuditLog.objects.count(), 4)
+
+    def test_list_as_anonymous(self):
+        self.client.logout()
+        response = self.client.get(self.url)
+        assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    def test_only_super_access(self):
+        self.force_login_user(User.objects.get(username='anotheruser'))
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_show_all_access_logs(self):
+        self.force_login_user(User.objects.get(username='admin'))
+        response = self.client.get(self.url)
+        self.assert_audit_log_results_equal(
+            response=response, expected_kwargs={'action': AuditAction.AUTH}
+        )
+
+    def test_search_by_username(self):
+        self.force_login_user(User.objects.get(username='admin'))
+        response = self.client.get(f'{self.url}?q=user__username:anotheruser')
+        another_user = User.objects.get(username='anotheruser')
+        self.assert_audit_log_results_equal(
+            response=response,
+            expected_kwargs={'action': AuditAction.AUTH, 'user': another_user},
+        )
+
+    def test_search_by_date(self):
+        another_user = User.objects.get(username='anotheruser')
+        with skip_login_access_log():
+            self.client.force_login(User.objects.get(username='admin'))
+        tomorrow = now() + timedelta(days=1)
```

Comment:
Not related to your code but we are trying to use `timezone.now()` (and import `from django.utils import timezone`) instead of just using `now()` which could be ambiguous. If you see former import, do not hesitate to correct them.